### PR TITLE
chore: Flatten Travis config

### PR DESCRIPTION
**Pull request by nschonni**
- Remove deprecated "sudo" option
- Use named matrix sets instead of conditions

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -1,47 +1,36 @@
...
+        - SASS_SASSC_PATH=$TRAVIS_BUILD_DIR/../sassc
+        - SASS_SPEC_PATH=~$TRAVIS_BUILD_DIR/sass-spec
+      before_script:
+        - git clone https://github.com/sass/libsass.git $SASS_LIBSASS_PATH
+        - (cd $SASS_LIBSASS_PATH; git checkout $GITISH)
+        - git clone https://github.com/sass/sassc.git $SASS_SASSC_PATH
+        - (cd $SASS_SASSC_PATH; git checkout $GITISH)
+        - make -C $SASS_SASSC_PATH
+      script: bundle exec sass-spec.rb --impl $IMPL -c $COMMAND;
+    - name: "Dart-Sass"
```
Comment (nex3) [inline .travis.yml]:
```suggestion
    - name: "Dart Sass"
```

File: .travis.yml [hunk 2]
```diff
@@ -1,47 +1,36 @@
...
+      env: IMPL=dart-sass
+      before_script:
+        - curl -o dart.zip "https://storage.googleapis.com/dart-archive/channels/dev/release/latest/sdk/dartsdk-linux-x64-release.zip"
```
Comment (nex3) [inline .travis.yml]:
```suggestion
        - curl -o dart.zip "https://storage.googleapis.com/dart-archive/channels/stable/release/latest/sdk/dartsdk-linux-x64-release.zip"
```

I'm not sure why I had this using Dart's dev channel before.

Comment (nschonni) [reply .travis.yml]: I tried using the `dart:` key for the built-in travis support, but it didn't seem to work when I tried it

Comment (nex3) [reply .travis.yml]: Yeah, that assumes you have a `pubspec.yaml` at the root of your project, and if you don't it'll break.

Comment (nschonni) [reply .travis.yml]: Gotcha, didn't really dig too far 😄

**Pull request merged**